torchgeo/datasets/spacenet1.py

```diff
@@ -0,0 +1,215 @@
+        crs: Optional[CRS] = None,
+        res: float = None,
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        download: bool = False,
+        api_key: str = None,
+    ) -> None:
+
+        self.root = root
+        self.transforms = transforms
+
+        if not self._check_integrity():
+            if download:
+                self._download(api_key)
+            else:
+                raise RuntimeError(
+                    "Dataset not found. You can use download=True to download it."
+                )
+
+        super().__init__(
+            root=os.path.join(root, self.foldername),
+            crs=crs,
+            res=res,
+            transforms=transforms,
+        )
+        self.files = self._load_files(os.path.join(root, self.foldername))
+
+    def _load_files(self, root) -> List[Dict[str, str]]:
+        files = []
+        images = glob.glob(os.path.join(root, "*", self.filename_glob))
+        images = sorted(images)
+        for imgpath in images:
+            rawpath = imgpath.replace("RGB.tif", "8Band.tif")
+            lbl_path = os.path.join(
+                os.path.dirname(rawpath) + "-labels", "labels.geojson"
+            )
+            files.append({"rgb": imgpath, "8band": rawpath, "label": lbl_path})
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        filename = os.path.join(path)
+        with rio.open(filename) as img:
+            array = img.read().astype(np.float32)
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            return tensor, img.transform
+
+    def _load_label(self, path: str, tfm: Affine, shape: Tuple[int, int]) -> Tensor:
+        """Rasterizes the dataset's labels (in geojson format).
+
+        Args:
+            path (str): path to the label
+            tfm (Affine): transform of corresponding image
+            shape (List[int, int]): shape of corresponding image
+
+        Returns:
+            Tensor: label tensor
+        """
+
+        with fiona.open(path) as src:
+            labels = [feature["geometry"] for feature in src]
+
+        if not labels:
+            mask_data = np.zeros(shape=shape)
+        else:
+            mask_data = rasterize(
+                labels,
+                out_shape=shape,
+                fill=0,  # nodata value
+                transform=tfm,
+                all_touched=False,
+                dtype=np.uint8,
+            )
+
+        mask: Tensor = torch.from_numpy(mask_data).long()  # type: ignore[attr-defined]
+
+        return mask
+
+    def __len__(self) -> int:
+        """Return the number of samples in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
```

Comment:
The `__getitem__` for `GeoDataset`s should take a bounding box query, not an integer index.